src/main/java/com/github/princesslana/eriscasper/gateway/Gateway.java

```diff
@@ -87,19 +88,21 @@ private void sequenceNumberSeen(Optional<SequenceNumber> seq) {
     seq.ifPresent(sid -> lastSeenSequenceNumber = Optional.of(sid));
   }
 
-  @SuppressWarnings("unchecked")
+  private Consumer<RxWebSocketEvent> warnOnClosing(String url, Optional<ShardPayload> shard) {
+    return evt -> {
+      if (evt instanceof RxWebSocketEvent.Closing || evt instanceof RxWebSocketEvent.Closed) {
+        LOG.warn("Websocket closing: url={}, shard={}, event={}", url, shard, evt);
+      }
+    };
+  }
+
   public Observable<Event> connect(String url, BotToken token, Optional<ShardPayload> shard) {
-    Observable<RxWebSocketEvent> websocketEvents =
-        ws.connect(String.format("%s?v=%s&encoding=%s", url, VERSION, ENCODING))
-            .flatMap(
-                e ->
-                    e instanceof RxWebSocketEvent.Closing || e instanceof RxWebSocketEvent.Closed
-                        ? Observable.error(
-                            new ErisCasperFatalException("Websocket closed unexpectedly: " + e))
-                        : Observable.just(e));
+    CompositeDisposable disposables = new CompositeDisposable();
 
     Observable<Payload> ps =
-        websocketEvents
+        ws.connect(String.format("%s?v=%s&encoding=%s", url, VERSION, ENCODING))
+            .doOnNext(warnOnClosing(url, shard))
+            .doFinally(disposables::dispose)
```

Comment:
Are we sure this captures and produces the hearbeat/identify etc... as expected? Do we have some manual testing to back up this change?